src/util/misc.rs

```diff
@@ -34,31 +80,46 @@ pub fn remove_expired(pastas: &mut Vec<Pasta>, args: &Args) {
             // keep
             true
         } else {
+            log::debug!("Removing expired pasta ID: {} ({})", p.id, p.id_as_animals(&args.hash_ids));
+            if p.expiration > 0 && p.expiration <= timenow {
+                log::debug!("  Reason: Expired (expiration: {}, now: {})", p.expiration, timenow);
+            }
+            if p.burn_after_reads > 0 && p.read_count >= p.burn_after_reads {
+                log::debug!("  Reason: Burn after reads limit reached ({}/{})", p.read_count, p.burn_after_reads);
+            }
+            if args.gc_days > 0 && p.last_read_days_ago() >= args.gc_days {
+                log::debug!("  Reason: GC limit reached (last read {} days ago, limit: {})", p.last_read_days_ago(), args.gc_days);
+            }
+            
             // remove from database
             delete(None, Some(p.id), args);
 
             // remove the file itself
             if let Some(file) = &p.file {
-                if fs::remove_file(format!(
+                let file_path = format!(
                     "{}/attachments/{}/{}",
                     args.data_dir,
                     p.id_as_animals(&args.hash_ids),
                     file.name()
-                ))
-                .is_err()
-                {
-                    log::error!("Failed to delete file {}!", file.name())
+                );
+                log::debug!("Attempting to delete file: {}", file_path);
```

Comment:
bit overkill, makes the function harder to read as the important bits are further apart. I think its enough to have the error when the delete fails.